### DISCUSSION TITLE
Floating-point precision conversion error

### DISCUSSION THREAD

**Discussion by ApacheWang** [nlohmann/json#2878] [Q&A] [answered]
#### Please describe the steps to reproduce the issue.

1. pls use this case to reproduce.
`std::string tmp = "{\n"
                      "                  \"compareType\":0,\n"
                      "                  \"feature\":100,\n"
                      "                  \"id\":8,\n"
                      "                  \"leftNodeID\":17,\n"
                      "                  \"rightNodeID\":18,\n"
                      "                  \"threshold\":[\n"
                      "                    0.2709993869\n"
                      "                  ]\n"
                      "                }";
    json js = json::parse(tmp);
    cout << js.dump() << endl;`
2. And the threshold change to 0.27099938690000003

#### Which compiler and operating system are you using?

- Compiler: ___
- Operating system: ___

#### Which version of the library did you use?

- [ ] latest release version 3.9.1
- [ ] other release - please state the version: ___
- [ ] the `develop` branch

#### If you experience a compilation error: can you [compile and run the unit tests](https://github.com/nlohmann/json#execute-unit-tests)?

- [ ] yes
- [ ] no - please copy/paste the error message below

**Comment by nlohmann** [comment, marked as answer]
This is expected behavior, see https://json.nlohmann.me/features/types/number_handling/#number-serialization
